src/api/construction.rs

```diff
@@ -101,7 +101,7 @@ impl ConstructionApi {
         Ok(ConstructionPreprocessResponse {
             options:              Some(
                 serde_json::to_value(&options)
-                    .map_err(|err| ApiError::JsonEncodingFailed("options".to_string(), err))?,
+                    .map_err(|err| ApiError::InternalServerError(anyhow::anyhow!(err)))?,
```

Comment:
Given that you have
```
InternalServerError(#[from] anyhow::Error)
```
can't you just return `anyhow::anyhow!(err))`?

But I'm a little skeptical on whether `#[from]` is a good idea here as it might silence the type system in places where we might not actually want it to be an "internal" error. So I actually think I prefer you to remove that and keep this as it is :)